lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -146,11 +144,10 @@ bool _hasNonComparableOperands(TypeSystem typeSystem, BinaryExpression node) {
       !(_isFixNumIntX(leftType) && _isCoreInt(rightType));
 }
 
-bool _isCoreInt(DartType type) =>
-    type.name == 'int' && type.element?.library?.name == _dartCoreLibraryName;
+bool _isCoreInt(DartType type) => type.isDartCoreInt;
 
 bool _isFixNumIntX(DartType type) =>
-    (type.name == 'Int32' || type.name == 'Int64') &&
+    (type.element?.name == 'Int32' || type.element?.name == 'Int64') &&
```

Comment:
I'd replaced this with `type is InterfaceType`, and then call `type.element.name` without `null` protection.
Maybe extract `element` into a local variable.

Reply:
👍